### PR TITLE
Fault domain helper

### PR DESCRIPTION
**Pull request by mellenburg**
This feature will allow users to name arbitrary regions within which one may declare num_private_agents, num_public_agents, and num_zones

See the updated README for more details: https://github.com/dcos/dcos-launch/blob/b897f0d0412f9fd61f3f1ec9709599cad488854f/dcos_launch/sample_configs/README.md

### CODE REVIEW

File: dcos_launch/config.py [hunk 1]
```diff
@@ -205,6 +205,21 @@ def get_validated_config(user_config: dict, config_dir: str) -> dict:
         'required': True}}
 
 
+def _validate_fault_domain_helper(field, value, error):
+    for fd in value:
+        pass
+
+    if not value.startswith('http'):
```
Comment (margaret) [inline dcos_launch/config.py]: Did you mean to put this inside the for loop above?

File: dcos_launch/config.py [hunk 2]
```diff
@@ -205,6 +205,21 @@ def get_validated_config(user_config: dict, config_dir: str) -> dict:
...
+        error(field, 'Not a valid HTTP URL')
+
+
+def _get_num_agents(doc: dict, field_name: str):
+    total = 0
+    for fd in doc.get('fault_domain_helper', list()):
+        total += fd[field_name]
+    return total
```
Comment (margaret) [inline dcos_launch/config.py]: `return sum([fd[field_name] for fd in doc.get('fault_domain_helper', list())] )`?

File: dcos_launch/config.py [hunk 3]
```diff
@@ -205,6 +205,21 @@ def get_validated_config(user_config: dict, config_dir: str) -> dict:
         'required': True}}
 
 
+def _validate_fault_domain_helper(field, value, error):
+    for fd in value:
+        pass
```
Comment (margaret) [inline dcos_launch/config.py]: was something supposed to be here?

File: dcos_launch/config.py (hunk 2 above)
Comment (margaret) [reply dcos_launch/config.py]: Also, this method doesn't seem to be called anywhere within the new changes 🤔

File: dcos_launch/config.py [hunk 4]
```diff
@@ -251,7 +274,30 @@ def get_validated_config(user_config: dict, config_dir: str) -> dict:
             'public_agent_list': {'readonly': True},
             'fault_domain_script_filename': {
                 'coerce': 'expand_local_path',
-                'excludes': 'fault_domain_script_contents'}}}}
+                'excludes': 'fault_domain_script_contents'}}},
+    'fault_domain_helper': {
+        'type': 'dict',
```
Comment (margaret) [inline dcos_launch/config.py:277]: should you put the `_validate_fault_domain_helper` here as the `'validator'`?

File: dcos_launch/config.py (hunk 2 above)
Comment (mellenburg) [reply dcos_launch/config.py]: heh, thanks for noticing!

File: dcos_launch/config.py (hunk 4 above)
Comment (mellenburg) [reply dcos_launch/config.py:277]: Yup! Fixed in most recent update

**Review by margaret — APPROVED** [review]

**Comment by orsenthil** [comment]
@mellenburg / @margaret  - was there a JIRA for this?

**Comment by orsenthil** [comment]
https://jira.mesosphere.com/browse/DCOS-18424

**Pull request merged**